lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1566,9 +1568,13 @@ LogicalResult FIRRTLLowering::visitDecl(WireOp op) {
   if (resultType.isInteger(0))
     return setLowering(op, Value());
 
-  // Name attr is requires on sv.wire but optional on firrtl.wire.
+  // Name attr is required on sv.wire but optional on firrtl.wire.
   auto nameAttr = op.nameAttr() ? op.nameAttr() : builder.getStringAttr("");
-  return setLoweringTo<sv::WireOp>(op, resultType, nameAttr);
+  auto symName = op.nameAttr() ? op.nameAttr() : StringAttr();
+  // This is not a temporary wire created by the compiler, so attach a symbol
+  // name.
+  // TODO: How to ensure there's no symbol name conflict ?
```

Comment:
The existing name is unique in the module, but not in the symbol table.  I would do something like "__<module_name>_<wirename>".  Checking is a bit tricky due to SymbolTable not looking for symbol definitions in nested regions.